src/programs/simulate/simulate.cpp

```diff
@@ -548,26 +594,87 @@ class SimulateApp : public MyApp {
             temp_potential += sp.ReturnScatteringParamtersA(atom_id, iGaussian) * fabsf(t0);
 
         } // loop over gaussian fits
+#endif
 
         return temp_potential *= this->lead_term;
     };
 
     // Shift the curves to the right as the values from Shang/Sigworth are distance to VDW radius (avg C/O/N/H = 1.48 A)
     // FIXME now that you are saving distances, you can also consider polar/non-polar residues separately for an "effective" distance since the curves have the same shape with a linear offset.
-    const float PUSH_BACK_BY = -1.48;
+    inline float return_hydration_weight_exact(float radius) {
+        return 0.5f + 0.5f * std::erff(((radius) - (HYDRATION_VALS[2] + xtra_shift * this->wanted_pixel_size)) / (sqrtf(2) * HYDRATION_VALS[5])) +
+               HYDRATION_VALS[0] * expf(-powf((radius) - (HYDRATION_VALS[3] + xtra_shift * this->wanted_pixel_size), 2) / (2 * powf(HYDRATION_VALS[6], 2))) +
+               HYDRATION_VALS[1] * expf(-powf((radius) - (HYDRATION_VALS[4] + xtra_shift * this->wanted_pixel_size), 2) / (2 * powf(HYDRATION_VALS[7], 2)));
+    }
+
+    inline float return_hydration_weight(float r) {
```

Comment:
Piecewise polynomial implementation of the hydration weight, based on a fit using splines to the "exact weights" used before. Difference should be numerically negligible.